### PR TITLE
[inductor] split test_cpp_wrapper.py into cpu and cuda test files

### PR DESCRIPTION
**Pull request by jgong5**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #115479
* #115167

cc @voznesenskym @penguinwu @EikanWang @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @peterbell10 @ipiszy @yf225 @chenyang78 @kadeng @muchulee8 @aakhundov @ColinPeppler

### CODE REVIEW

**Comment by atalman** [comment]
@jgong5 Please rebase your changes with https://github.com/pytorch/pytorch/pull/115430 . This should fix all the failures related to running CPU tests on CUDA devices

**Comment by jgong5** [comment]
> @jgong5 Please rebase your changes with #115430 . This should fix all the failures related to running CPU tests on CUDA devices

Done.

File: test/inductor/test_cpu_cpp_wrapper.py [hunk 1]
```diff
@@ -0,0 +1,312 @@
...
+    if condition:
+        assert device == "cpu"
+        setattr(
+            CppWrapperTemplate,
+            test_name,
+            fn,
+        )
+
+
+class BaseTest(NamedTuple):
```
Comment (atalman) [inline test/inductor/test_cpu_cpp_wrapper.py:125]: We would need to add condition here:
```
if RUN_CPU:
``

File: test/inductor/test_cuda_cpp_wrapper.py [hunk 2]
```diff
@@ -0,0 +1,237 @@
...
+    fn.__dict__ = copy.deepcopy(func.__dict__)
+    if condition:
+        assert device == "cuda"
+        setattr(
+            CudaWrapperTemplate,
+            test_name,
+            fn,
+        )
+
+
```
Comment (atalman) [inline test/inductor/test_cuda_cpp_wrapper.py:145]: Alco conditon here
```
if RUN_CUDA:
```

File: test/inductor/test_cuda_cpp_wrapper.py [hunk 3]
```diff
@@ -0,0 +1,237 @@
...
+        finally:
+            tests.tearDown()
+            tests.tearDownClass()
+
+    fn.__name__ = test_name
+    import copy
+
+    fn.__dict__ = copy.deepcopy(func.__dict__)
+    if condition:
+        assert device == "cuda"
```
Comment (atalman) [inline test/inductor/test_cuda_cpp_wrapper.py:138]: Why this assert needed ?

File: test/inductor/test_cpu_cpp_wrapper.py [hunk 4]
```diff
@@ -0,0 +1,312 @@
...
+        finally:
+            tests.tearDown()
+            tests.tearDownClass()
+
+    fn.__name__ = test_name
+    import copy
+
+    fn.__dict__ = copy.deepcopy(func.__dict__)
+    if condition:
+        assert device == "cpu"
```
Comment (atalman) [inline test/inductor/test_cpu_cpp_wrapper.py:117]: I don't think this assert needed

**Comment by atalman** [comment]
HI @jgong5 to fix this failure: https://github.com/pytorch/pytorch/actions/runs/7190312080/job/19583808375?pr=115479#step:16:3011
```
AttributeError: 'ForeachTests' object has no attribute 'test_foreach_cpp_wrapper_cuda'. Did you mean: 'test_foreach_cpp_wrapper'?
```
Some tests needs to be renamed, please take a look at this commit:
https://github.com/pytorch/pytorch/pull/115651/commits/2b3314e3dc4e0e52a54a8ef2fb94c8b26d2c7cc7

**Review by atalman — APPROVED** [review]
lgtm

**Comment by desertfire** [comment]
will land via https://github.com/pytorch/pytorch/pull/115928

**Comment by desertfire** [comment]
> will land via #115928

never mind. I will reland from here.

**Pull request closed without merging**